openupgrade_scripts/scripts/survey/14.0.3.2/pre-migration.py

```diff
@@ -0,0 +1,234 @@
+    ("survey.survey_expired", "survey.survey_closed_expired"),
+    ("survey.survey_void", "survey.survey_void_content"),
+    ("survey.survey_print", "survey.survey_page_print"),
+    ("survey.question", "survey.question_container"),
+    ("survey.back", "survey.survey_button_form_view"),
+    ("survey.survey", "survey.survey_page_fill"),
+    ("survey.page", "survey.survey_fill_form"),
+    ("survey.survey_header", "survey.survey_fill_header"),
+    ("survey.survey_init", "survey.survey_fill_form_start"),
+    ("survey.sfinished", "survey.survey_fill_form_done"),
+    ("survey.result", "survey.survey_page_statistics"),
+]
+
+
+def fill_survey_survey_session_state(env):
+    # done in pre-migration because this field is used in some computes
+    openupgrade.map_values(
+        env.cr,
+        openupgrade.get_legacy_name("state"),
+        "state",
+        [("skip", "in_progress")],
+        table="survey_user_input",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE survey_survey
+        ADD COLUMN session_state varchar""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE survey_survey
+        ADD COLUMN session_start_time timestamp without time zone""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE survey_survey ss
+        SET session_state = CASE WHEN sui.state = 'new' THEN 'ready'
+            WHEN sui.state = 'in_progress' THEN 'in_progress'
+            ELSE NULL END, session_start_time = CASE
+            WHEN sui.state IN ('new', 'in_progress') THEN ss.write_date
+            ELSE NULL END
+        FROM survey_user_input sui
+        WHERE sui.survey_id = ss.id AND ss.state != 'draft'""",
+    )
+
+
+def fast_fill_survey_input_some_values(env):
+    # some new computes depend of this is_session_answer field
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE survey_user_input
+        ADD COLUMN is_session_answer boolean""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE survey_user_input
+        ADD COLUMN nickname varchar""",
+    )
+    # is_session_answer
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE survey_user_input sui
+        SET is_session_answer = TRUE, start_datetime = sui.write_date
+        FROM survey_survey ss
+        WHERE sui.survey_id = ss.id AND ss.session_state IN ('ready', 'in_progress')""",
+    )
+    # nickname
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE survey_user_input sui
+        SET nickname = rp.name
+        FROM survey_survey ss, res_partner rp
+        WHERE sui.survey_id = ss.id AND sui.partner_id = rp.id""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE survey_user_input sui
+        SET nickname = sui.email
+        FROM survey_survey ss
+        WHERE sui.survey_id = ss.id AND sui.nickname IS NULL""",
```

Comment:
Same question about the join on survey survey here.

Reply:
survey_survey is the FROM, what do you mean?

Reply:
deleted unused table